chemprop/models/model.py

```diff
@@ -69,8 +69,10 @@ def __init__(
         X_d_transform: ScaleTransform | None = None,
     ):
         super().__init__()
-
-        self.save_hyperparameters(ignore=["message_passing", "agg", "predictor"])
+        # manually add X_d_transform to hparams to suppress lightning's warning about double saving
+        # its state_dict values.
+        self.save_hyperparameters(ignore=["X_d_transform", "message_passing", "agg", "predictor"])
+        self.hparams["X_d_transform"] = X_d_transform
         self.hparams.update(
             {
                 "message_passing": message_passing.hparams,
```

Comment:
```suggestion
        self.hparams.update(
            {
                "X_d_transform": X_d_transform,
                "message_passing": message_passing.hparams,
```

Reply:
Thank you for the suggestion. I also thought about including X_d_transform in the update, but decided it would be better to do separately to match what is done in message passing base with V_d_transform for example. Having it separate also helps show that it doesn't have its own hparams like message passing does. I'll go ahead and set this to merge because you approved it. If you want to talk about this suggested change more, you could open another PR with it.

Reply:
I see! How about not using `hparams.update` at all? I don't think it's necessary to separate these two parts here, as it doesn't provide any new context. If you think it's worth mentioning that the message passing has its own `hparams`, it would be clearer to add a comment in the code instead of using a different style to separate them.